### PR TITLE
Update anti-e tau discriminant

### PR DESCRIPTION
**Pull request by mbluj**
This PR is to update an MVA anti-e tau discriminant as preparation to introduction of discriminants with a new training with current data. The updated consists of
* values of a set of MVA input parameters are constrained against unphysical outliers (key change for the new MVA training),
* making `usePhiAtEcalEntranceExtrapolation` parameter an internal parameter of the MVA algorithm to which it belongs rather than of the producer which calls the algorithm,
* a few other fixes to the code, e.g. range-based loops, a modern pointer instead of bare C pointer, etc.

The changes are not expected to change result, i.e. returned value of the MVA anti-e tau discriminant (together with related WPs) with the current training, or the changes will be marginal, i.e. below precision of performed tests with ~100k of ZLL (L=e,mu,tau) events.

A PR with additional changes in configuration files to introduce the new training will be send when the training fully validated and related payloads loaded to CondDB.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoTauTag/RecoTau/python/PATTauDiscriminationAgainstElectronMVA6_cfi.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 patTauDiscriminationAgainstElectronMVA6 = cms.EDProducer("PATTauDiscriminationAgainstElectronMVA6",
     # tau collection to discriminate
-    PFTauProducer = cms.InputTag('slimmedTaus'),
+    PATTauProducer = cms.InputTag('slimmedTaus'),
```
[the remark this answers is not in the document]
Comment (mbluj) [reply RecoTauTag/RecoTau/python/PATTauDiscriminationAgainstElectronMVA6_cfi.py]: Yes, it is a bug fix. There is not effect expected in standard workflows, as the against-e MVA6 discriminant is not run using pat::Tau as input. The `PATTauDiscriminationAgainstElectronMVA6` plugin has been developed to rerun the discriminant with a new training either by users or as a part of Re-MiniAOD workflow. But, the plugin has not been used as any training good enough to replace currently one has been provided.

**Comment by perrotta** [comment]
+1
- Mostly technical updates and code cleaning
- Only change that could possibly remotely affect outputs: protections added to avoid crashes in RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc
- No effect is visible in jenkins outputs, anyhow

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**